src/recipe/Baked.Recipe.Service.Application/CodeGeneration/CodeGenerationLayer.cs

```diff
@@ -1,46 +1,107 @@
+        yield return new Compile(_location);
+        yield return new CreateFiles(_location);
     }
 
-    public class GenerateCode(IGeneratedAssemblyCollection _generatedAssemblies)
+    public class GenerateCode(string _location, IGeneratedAssemblyCollection _generatedAssemblies, IGeneratedFileCollection _generatedFiles)
         : PhaseBase<DomainModel>(PhaseOrder.Early)
     {
         protected override void Initialize(DomainModel _)
         {
+            if (Directory.Exists(_location))
+            {
+                Directory.Delete(_location, true);
+            }
+
+            Directory.CreateDirectory(_location);
+
             Context.Add(_generatedAssemblies);
+            Context.Add(_generatedFiles);
         }
     }
 
-    public class Compile()
-        : PhaseBase<IGeneratedAssemblyCollection>(PhaseOrder.Latest)
+    public class Compile(string _location)
+        : PhaseBase<IGeneratedAssemblyCollection>(PhaseOrder.Late)
     {
         protected override void Initialize(IGeneratedAssemblyCollection generatedAssemblies)
         {
-            var provider = new GeneratedAssemblyProvider();
             foreach (var descriptor in generatedAssemblies)
             {
-                var assembly = new Compiler(descriptor).Compile();
+                new Compiler(descriptor).Compile(_location, $"Baked.g.{descriptor.Name}");
+            }
+        }
+    }
+
+    public class CreateFiles(string _location)
+        : PhaseBase<IGeneratedFileCollection>(PhaseOrder.Latest)
+    {
+        protected override void Initialize(IGeneratedFileCollection generatedFiles)
+        {
+            foreach (var descriptor in generatedFiles)
+            {
+                using var file = new FileStream(Path.Combine(_location, $"{descriptor.Name}.{descriptor.Extension}"), FileMode.Create);
+                file.Write(Encoding.UTF8.GetBytes(descriptor.Content));
+            }
+        }
+    }
 
-                provider.Add(descriptor.Name, assembly);
+    public class LoadGeneratedAssemblies(string _location)
+        : PhaseBase(PhaseOrder.Early)
+    {
+        protected override void Initialize()
+        {
+            var directoryfiles = Directory.GetFiles(_location);
+
+            var provider = new GeneratedAssemblyProvider();
+            var fileProvider = new GeneratedFileProvider();
+            foreach (var path in directoryfiles)
+            {
+                if (path.Contains("Baked.g"))
+                {
+                    provider.Add(Path.GetFileName(path).Replace("Baked.g.", string.Empty).Replace(".dll", string.Empty), Assembly.LoadFile(path));
+                }
+                else
+                {
+                    fileProvider.Add(Path.GetFileName(path)[..Path.GetFileName(path).LastIndexOf('.')], path);
+                }
             }
 
             Context.Add(provider);
+            Context.Add(fileProvider);
```

Comment:
consider reducing number phases, and reusing existing phases from other core layers where possible